### PR TITLE
#3: Format as YYYY-MM-DD to avoid 422 error from API

### PR DESCRIPTION
**Pull request by surdaft**
### What does this PR do?

- Experience using this project has shown that a 422 error is returned when the full RFC3339 date is provided to the time-offs endpoint.

### What is the effect of this change to users?

None

### How does it look like?

N/A

### Any background context you can provide?

I raised issue #3 regarding this

### What is needed from the reviewers?

Just a glance and tick. :) 

### Do the docs need to be updated?

No

### Should this change be mentioned in the release notes?

Unlikely, up to you guys

- [ ] CHANGELOG.md has been updated (if it exists)

### CODE REVIEW

File: v1/personio.go [hunk 1]
```diff
@@ -443,10 +443,10 @@ func (personio *Client) GetTimeOffs(start *time.Time, end *time.Time, offset int
 
 		query := req.URL.Query()
 		if start != nil {
-			query.Add("start_date", start.Format(time.RFC3339))
+			query.Add("start_date", start.Format("2006-01-02"))
```
Comment (lyind) [inline v1/personio.go:446]: Would you like to:
* make this a package private variable at the top of this file (eg. `const dateFormat = "2006-01-02"`)
* then also use that variable in `personio_test.go` (lines 103, 109)
?

**Review by lyind — CHANGES_REQUESTED** [review]

**Comment by lyind** [comment]
Hi @surdaft, awesome fix, would you like to change the points I mentioned in the review?


Also please read the below and leave "us" (Giant Swarm GmbH) a comment stating that you have read and agree to terms.

By opening this Pull Request, you certify that you:

- [ ] are the sole author of your contribution and are legally entitled to grant us a license to use it
  - or, your contribution is on behalf of, and with explicit permission of, the legal owner of the contribution
- [ ] grant us a nonexclusive, irrevocable, transferrable license to use, reproduce, display, sell, and further license your contribution and derivative works
- [ ] agree that your contribution is made freely and bears no obligation or expectation for payment, royalties, recognition, or other compensation from us
- [ ] agree for your contribution to be bound by the terms in the license accompanying this repository

If you are making this submission on behalf of a current or former employer, you also certify that you have authorization from the employer to make this contribution, or the employer has released all legal intrerest in this contribution.

File: v1/personio.go (hunk 1 above)
Comment (lyind) [reply v1/personio.go:446]: Maybe also add something like the following to the CHANGELOG.md under section `## [Unreleased]`:

```
### Fixed

- Fixed `GetTimeOffs()` date argument format (YYYY-MM-DD)
```

**Comment by surdaft** [comment]
> Also please read the below and leave "us" (Giant Swarm GmbH) a comment stating that you have read and agree to terms.

I have read and agree to each of these terms.

File: v1/personio.go [hunk 2]
```diff
@@ -17,6 +17,8 @@ const DefaultBaseUrl = "https://api.personio.de/v1"
 
 const timeOffsMaxLimit = 200
 
+const QUERY_DATE_FORMAT = "2006-01-02"
```
Comment (lyind) [inline v1/personio.go:20]: Could you please rename this to `queryDateFormat` to make it a package private constant and comply to the golang naming convention?

**Review by lyind — CHANGES_REQUESTED** [review]
Variable name change to comply to conventions and avoid exporting the variable as part of the package's interface.

Looks good otherwise.

File: v1/personio.go (hunk 2 above)
Comment (surdaft) [reply v1/personio.go]: This has now been done. :)

**Pull request merged**